numpy/core/tests/test_multiarray.py

```diff
@@ -8427,11 +8427,14 @@ def test_to_int_scalar(self):
             assert_equal(5, int_func(np.bytes_(b'5')))
             assert_equal(6, int_func(np.unicode_(u'6')))
 
-            class HasTrunc:
-                def __trunc__(self):
-                    return 3
-            assert_equal(3, int_func(np.array(HasTrunc())))
-            assert_equal(3, int_func(np.array([HasTrunc()])))
+            # The delegation of int() to __trunc__ was deprecated in
+            # Python 3.11.
+            if sys.version_info < (3, 11):
+                class HasTrunc:
+                    def __trunc__(self):
+                        return 3
+                assert_equal(3, int_func(np.array(HasTrunc())))
+                assert_equal(3, int_func(np.array([HasTrunc()])))
```

Comment:
I would recommend adding an `else: pass` here, because that would enable the block to be automatically discovered and cleaned up with `pyupgrade` in the future (see https://github.com/asottile/pyupgrade#python2-and-old-python3x-blocks)

Reply:
@felixxm Could you implement this?

Reply:
your call. Pyupgrade doesn't need to be used continuously to be valuable, but I guess leaving these versioned branches around even after they become obsolete doesn't hurt so there's no strong incentive here.

Reply:
Added